java/arcs/sdk/wasm/WasmCombinedSingletonHandle.kt

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.sdk.wasm
+
+/** Combined Handle to allow events on multiple handles to trigger actions. */
+class WasmCombinedSingletonHandle<T : WasmEntity>(val handles: List<WasmSingletonImpl<T>>) {
```

Comment:
I would suggest we consider a way to register onUpdate callbacks for handles that do not have matching types, e.g.
```
onUpdate(handles.a, handles.b, handles.c) { aVal, bVal, cVal ->
  (...)
}
```

WDYT?

I believe this would require have one override per a number of arguments, but it's much more flexible that limiting the combine to handles of the same type.